src/main/java/at/hannibal2/skyhanni/features/skillprogress/SkillProgress.kt

```diff
@@ -0,0 +1,459 @@
+        }
+    }
+
+    private fun renderDisplay() {
+        when (val textAlignment = config.textAlignmentProperty.get()) {
+            SkillProgressConfig.TextAlignment.NONE -> {
+                config.displayPosition.renderStringsAndItems(listOf(display), posLabel = "Skill Progress")
+            }
+
+            SkillProgressConfig.TextAlignment.CENTERED,
+            SkillProgressConfig.TextAlignment.LEFT,
+            SkillProgressConfig.TextAlignment.RIGHT -> {
+                config.displayPosition.renderRenderables(
+                    listOf(Renderable.fixedSizeLine(horizontalContainer(display, textAlignment.alignment), maxWidth)),
+                    posLabel = "Skill Progress")
+            }
+
+            else -> {}
+        }
+    }
+
+    private fun renderBar() {
+        val progress = if (barConfig.useTexturedBar.get()) {
+            val factor = (skillExpPercentage.toFloat().coerceAtMost(1f)) * 182
+            maxWidth = 182
+            Renderable.texturedProgressBar(factor,
+                Color(SpecialColour.specialToChromaRGB(barConfig.barStartColor)),
+                texture = barConfig.texturedBar.usedTexture.get(),
+                useChroma = barConfig.useChroma.get())
+
+        } else {
+            maxWidth = barConfig.regularBar.width
+            Renderable.progressBar(skillExpPercentage,
+                Color(SpecialColour.specialToChromaRGB(barConfig.barStartColor)),
+                Color(SpecialColour.specialToChromaRGB(barConfig.barStartColor)),
+                width = maxWidth,
+                height = barConfig.regularBar.height,
+                useChroma = barConfig.useChroma.get())
+        }
+
+        config.barPosition.renderRenderables(listOf(progress), posLabel = "Skill Progress Bar")
+    }
+
+    @SubscribeEvent
+    fun onProfileSwitch(event: PreProfileSwitchEvent) {
+        display = emptyList()
+        allDisplay = emptyList()
+        etaDisplay = emptyList()
+        skillExpPercentage = 0.0
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+        if (lastUpdate.passedSince() > 3.seconds) showDisplay = config.alwaysShow.get()
+
+        if (event.repeatSeconds(1)) {
+            allDisplay = formatAllDisplay(drawAllDisplay())
+            etaDisplay = drawETADisplay()
+        }
+
+        if (event.repeatSeconds(2)) {
+            update()
+            updateSkillInfo(activeSkill)
+        }
+    }
+
+    @SubscribeEvent
+    fun onLevelUp(event: SkillOverflowLevelupEvent) {
+        if (!isEnabled()) return
+        if (!config.overflowConfig.enableInChat) return
+        val skillName = event.skill.displayName
+        val oldLevel = event.oldLevel
+        val newLevel = event.newLevel
+        val skill = SkillAPI.storage?.get(event.skill) ?: return
+        val goalReached = newLevel == skill.customGoalLevel && customGoalConfig.enableInChat
```

Comment:
Why do we send two times something in chat about goal reached? is this intended?